modules/core/src/main/java/org/apache/synapse/config/xml/PayloadFactoryMediatorSerializer.java

```diff
@@ -148,8 +163,23 @@ public OMElement serializeSpecificMediator(Mediator m) {
         return payloadFactoryElem;
     }
 
+    private boolean isFreeMarkerTemplate(PayloadFactoryMediator mediator) {
+
+        return mediator.getTemplateType() != null && mediator.getTemplateType().equalsIgnoreCase(FREEMARKER);
+    }
     public String getMediatorClassName() {
         return PayloadFactoryMediator.class.getName();
     }
 
+    public static String removeCDATAFromPayload(String inputPayload) {
+        if (inputPayload.startsWith("<![CDATA[")) {
+            inputPayload = inputPayload.substring(9);
+            int i = inputPayload.indexOf("]]>");
```

Comment:
since the index of returns, the first occurrence is there a chance this will break in CDATA inside CDATA scenario?

Reply:
fixed this